NMS-13364: Table formatting issues in new docs (#3531)
OpenNMS/opennms: docs/modules/operation/pages/service-assurance/monitors/SnmpMonitor.adoc

> The SNMP monitor gives a generic possibility to monitor states and results from SNMP agents.

Reviewer: ```suggestion
The SNMP monitor provides a common platform to monitor states and results from SNMP agents.
```

* Test multiple values in a whole table

To decide which mode to use, use the `walk` and `match-all` parameters.

The _Operating mode selection_ and _Monitor specific parameters for the SnmpMonitor_ tables below provide more information about these operation modes.

.Operating mode selection
[options="header"]
[cols="1,1,3"]
|===
| walk       | match-all                                                  | Operating mode
.3+|true      | true                                                     | tabular, all values must match
| false    | tabular, any value must match
    | count    | specifies that the value of at least minimum and at most
                   maximum objects encountered in
.3+|false     | true                                                     | scalar
    | false    | scalar
    | count    | tabular, between minimum and maximum values must match
|===

== Monitor facts

[options="autowidth"]
|===
| Class Name     | `org.opennms.netmgt.poller.monitors.SnmpMonitor`
| Remote Enabled | false
|===

When the monitor is configured to persist the response time, it counts the total amount of time spent until it obtains a successful one, including the retries.

> It will not store the time spent during the last successful attempt.

Reviewer: ```suggestion
It will store the entire poll time, not just the time spent during the last successful attempt.
```

                      `!=`: Not equal to. Applied in numeric context if both operand and observed object value are

>                             numeric, otherwise in string context as a case-sensitive exact match. +

Reviewer: ```suggestion
                            numeric, otherwise in string context as a case-sensitive exact non-match. +
```

|===

This monitor implements the <<service-assurance/monitors/introduction.adoc#ga-service-assurance-monitors-common-parameters, Common Configuration Parameters>>.

.Variables that can be used in the reason-template parameter
[options="header"]
[cols="1,3"]
|===
| Variable           | Description
| $\{hex}           | Value of the `hex` parameter.
| $\{ipaddr}        | IP address polled.
| $\{matchall}      | Value of the `match-all` parameter.
| $\{matchcount}    | When `match-all` is set to `count`, contains the number of matching instances encountered.
| $\{maximum}       | Value of the `maximum` parameter.
| $\{minimum}       | Value of the `minimum` paramater.
| $\{observedvalue} | Polled value that made the monitor succeed or fail.
| $\{oid}           | Value of the `oid` parameter.
| $\{operand}       | Value of the `operand` parameter.
| $\{operator}      | Value of the `operator` parameter.
| $\{port}          | Value of the `port` parameter.
| $\{retry}         | Value of the `retry` parameter.
| $\{timeout}       | Value of the `timeout` parameter.
| $\{walk}          | Value of the `walk` parameter.
|===

> == Example for monitoring scalar object

Reviewer: ```suggestion
== Example: monitoring a scalar object
```

<monitor service="HP-Insight-Drive-Physical" class-name="org.opennms.netmgt.poller.monitors.SnmpMonitor" />
----
<1> OID for SNMP table with all physical drive states.
<2> Enable walk mode to test every entry in the table against the test criteria.
<3> Test operator for integer.
<4> Integer 2 as operand for the test.
<5> Test in walk mode has to pass for every entry in the table.
<6> Encode MIB status in the reason code to give more detailed information if the service goes down.

> == Example: test SNMP table with all matching values

Reviewer: ```suggestion
== Example: test SNMP table with some matching values
```
